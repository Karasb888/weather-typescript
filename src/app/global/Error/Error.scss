.errorText {
    font-size: 30px;
    font-family: 'Cooper-Hewitt';
    font-weight: 600;
    color: #ff0000bd;
}

.errorIcon {
    margin-bottom: 20px;
}

.errorContainer {
    display: flex;
    padding: 30px 0;
    flex-direction: column;
    align-items: center;
}