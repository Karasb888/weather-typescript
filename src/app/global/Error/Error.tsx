import * as React from 'react';
import ErrorIcon from '../../icons/ErrorIcon';
import * as styles from './Error.scss';
import { Typography } from '@material-ui/core';

interface Props {
    errorText: string;
};

const Error: React.FC<Props> = ({ errorText }) => {
    return (
        <div className={styles.errorContainer}>
            <div className={styles.errorIcon}>
                <ErrorIcon
                    width="70"
                    height="70"
                    viewBox="0 0 512 512"
                    svgClass="error-icon"
                />
            </div>
            <Typography variant="h4" color="error" align="center">{errorText}</Typography>
        </div>
    );
};

export default Error;
