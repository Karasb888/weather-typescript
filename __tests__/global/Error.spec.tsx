import * as React from 'react';
import { shallow } from 'enzyme';
import Error, { Props } from '../../src/app/global/Error/Error';

const exampleProps: Array<Props> = [{errorText: 'text', errorSize: 'large'}, {errorText: 'text2222', errorSize: 'small'}];

describe('++++ <Error /> ++++', () => {
    exampleProps.forEach((errorProp) => {
        it('Renders correctly with different props', () => {
            const wrapper = shallow(<Error {...errorProp} />);
            expect(wrapper).toMatchSnapshot();
        });
    })
});
