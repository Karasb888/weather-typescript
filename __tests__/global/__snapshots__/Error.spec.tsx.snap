// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <Error /> ++++ Renders correctly with different props 1`] = `
<div
  className="errorContainer"
>
  <div
    className="errorIcon"
  >
    <ErrorIcon
      height="70"
      svgClass="error-icon"
      viewBox="0 0 512 512"
      width="70"
    />
  </div>
  <WithStyles(ForwardRef(Typography))
    align="center"
    color="error"
    variant="h4"
  >
    First example text
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`++++ <Error /> ++++ Renders correctly with different props 2`] = `
<div
  className="errorContainer"
>
  <div
    className="errorIcon"
  >
    <ErrorIcon
      height="70"
      svgClass="error-icon"
      viewBox="0 0 512 512"
      width="70"
    />
  </div>
  <WithStyles(ForwardRef(Typography))
    align="center"
    color="error"
    variant="h4"
  >
    Second example text
  </WithStyles(ForwardRef(Typography))>
</div>
`;
